cm/README.md

```diff
@@ -18,14 +11,202 @@ It must contain:
+def fun(self, selection: dict, rasters: list, params: dict):
+    """New cm description"""
+    result = process(selection, rasters, params)
+    return result
+
+
+if __name__ == "__main__":
+    cm_base.start_app(app)
+````
+
+The selected area and the selected raster 
+are in the variable ```selection``` and ```rasters``` respectively.
+
+The variable ```params``` refers to the data provided by the form on the frontend.
+
+## Test.py
+
+Below is an example of cm test.
+
+```python
+import unittest
+
+from new_cm import process
+
+selection = "selection.shp"
+raster = "raster.tif"
+params = 10
+
+class TestNewCM(unittest.TestCase):
+    def test_new_cm_feature(self):
+        result = process(selection, rasters, params)
+        self.assertIsNotNone(result)
+
+
+if __name__ == "__main__":
+    unittest.main()
+```
+
+## Schema.json
+
+Below is an example of the data schema expected by the cm.
+
+```
+{
+  "type": "object",
+  "properties": {
+    "factor": {
+      "type": "number",
+      "default": 1
+    }
+  }
+}
+```
+
+## Requirements.txt
+
+Below is an example of the dependencies used by a cm.
+
+```
+lxml==4.6.2
+GDAL==3.0.4
+```
+
+Only the dependencies specific to the cm are to be mentioned in the requirements.txt.
+
+The dependencies below will be available for all cm.
+
+````
+marshmallow==3.9.1
+marshmallow-union==0.1.15.post1
+jsonschema==3.2.0
+celery==5.0.2
+redis==3.5.3
+requests
+````
+
+Whenever possible, it is preferable to specify 
+the version of the dependency used. 
+
+## Docker
+
+### Dockerfile
+
+Below is an example of a Dockerfile
+
+```
+FROM ubuntu:20.04
+RUN apt-get update && \
+    apt-get --yes install python3 python3-pip &&\
+    rm -rf /var/cache/apt/archives/
+WORKDIR cm-first_cm
+COPY first_cm/requirements.txt requirements.txt
+RUN pip3 install -r requirements.txt
+COPY base /tmp/base
+RUN cd /tmp/base && pip3 install . && python3 test.py
+COPY first_cm .
+RUN python3 test.py
+CMD ["python3", "worker.py"]
+```
+
+### Docker-compose
 
 After adding those entrypoint, you can add your calculation module in the docker-compose.yml file at the root of this directory.
 
+
+Below is an example of the creation of the new service.
+
+```
+cm-first_cm:
+  build:
+    context: ./cm
+    dockerfile: first_cm/Dockerfile
+  environment:
+    UPLOAD_DIR: /upload-dir
+  volumes:
+    - upload-data:/upload-dir
+  env_file:
+    .env
+```
+
+# Test a cm
+
+Currently to test a particular cm, you will need to first start enermaps using docker-compose up in the root directory.
+Then you can connect to the redis queue. The multiply cm is given as an example.
+
+We strongly encourage you to write unittest that verify the good working of a calculation module separated from the enermaps api.
+
 # CMs interaction
 
 Currently, the cm registers on the redis queue trough the celery library.
 An helper library is given, you can find it under base/BaseCM/ and an example is given in
```

Comment:
Shouldn't we also make available the `first_cm` that is used in this doc?

Reply:
the first_cm repo does not exist, it just was use as example 
I've changed first_cm into new_cm to get it clearer